src/coreComponents/linearAlgebra/DofManager.cpp

```diff
@@ -814,12 +825,19 @@ void DofManager::setSparsityPatternOneBlock( SparsityPatternView< globalIndex >
       makeConnLocPattern< LOC, CONN >( mesh,
                                        rowField.key,
                                        rowField.numComponents,
+                                       rowField.globallyCoupledComponents, // select only globally coupled components
                                        rowField.numGlobalDof,
                                        regions,
                                        connLocRow );
     } );
 
-    if( colFieldIndex == rowFieldIndex )
+    // we create a temporary mask including all components
+    // if the colFieldIndex is equal to the rowFieldIndex and all components are in the mask,
+    // we don't need to recomponent the connLocPattern
+    CompMask const allComponentsMask( colField.numComponents, true );
+    bool const areAllComponentsCoupled = colField.globallyCoupledComponents.begin() == allComponentsMask.begin();
```

Comment:
It's unfortunate that I added `operator==` only to the `ComponentMask::Iterator` but not `ComponentMask` itself. This code here is fully correct though, just mentioning in case it raises someone's eyebrow.